Tests/PointFreeTests/EitherIOTests.swift

```diff
@@ -30,4 +30,15 @@ class EitherIOTests: TestCase {
 
     XCTAssertTrue(result.isRight)
   }
+
+  func testRetry_MaxRetriesZero() {
+    let thing = EitherIO<Prelude.Unit, Prelude.Unit>(run: IO {
+      return .right(unit)
+    })
+      .retry(maxRetries: 0)
```

Comment:
I just wanted to make sure the degenerate case of `maxRetries: 0` works as expected.